libsodium-sys/build.rs

```diff
@@ -142,186 +123,44 @@ fn find_libsodium_pkg() {
-        let opt_install_path = if entry_path.starts_with("include") {
-            let is_dir = (entry.unix_mode().unwrap() & S_IFDIR as u32) != 0;
-            if is_dir {
-                let _ = fs::create_dir(&Path::new(&install_dir).join(entry_path));
-                None
-            } else {
-                Some(Path::new(&install_dir).join(entry_path))
-            }
-        } else if entry_path == unpacked_lib {
-            Some(lib_install_dir.join("libsodium.lib"))
-        } else {
-            None
-        };
-        if let Some(full_install_path) = opt_install_path {
-            let mut buffer = Vec::with_capacity(entry.size() as usize);
-            assert_eq!(entry.size(), entry.read_to_end(&mut buffer).unwrap() as u64);
-            let mut file = File::create(&full_install_path).unwrap();
-            file.write_all(&buffer).unwrap();
-        }
-    }
-
-    println!("cargo:rustc-link-lib=static=libsodium");
-    println!(
-        "cargo:rustc-link-search=native={}",
-        lib_install_dir.display()
-    );
+fn get_install_dir() -> PathBuf {
+    PathBuf::from(env::var("OUT_DIR").unwrap()).join("libsodium_installed/")
 }
 
-#[cfg(all(windows, not(target_env = "msvc")))]
 fn build_libsodium() {
-    use flate2::read::GzDecoder;
-    use tar::Archive;
-
-    // Download gz tarball
-    let install_dir = get_install_dir();
-    let lib_install_dir = Path::new(&install_dir).join("lib");
-    fs::create_dir_all(&lib_install_dir).unwrap();
-    let url = format!("{}libsodium-{}-mingw.tar.gz", DOWNLOAD_BASE_URL, VERSION);
-    let compressed_file = download(&url, SHA256);
-
-    // Unpack the tarball
-    let gz_decoder = GzDecoder::new(compressed_file);
-    let mut archive = Archive::new(gz_decoder);
-
-    // Extract just the appropriate version of libsodium.a and headers to the install path
-    let arch_path = if cfg!(target_pointer_width = "32") {
-        Path::new("libsodium-win32")
-    } else if cfg!(target_pointer_width = "64") {
-        Path::new("libsodium-win64")
-    } else {
-        panic!("target_pointer_width not 32 or 64")
-    };
+    // Download sources
 
-    let unpacked_include = arch_path.join("include");
-    let unpacked_lib = arch_path.join("lib\\libsodium.a");
-    let entries = archive.entries().unwrap();
-    for entry_result in entries {
-        let mut entry = entry_result.unwrap();
-        let entry_path = entry.path().unwrap().to_path_buf();
-        let full_install_path = if entry_path.starts_with(&unpacked_include) {
-            let include_file = entry_path.strip_prefix(arch_path).unwrap();
-            Path::new(&install_dir).join(include_file)
-        } else if entry_path == unpacked_lib {
-            lib_install_dir.join("libsodium.a")
-        } else {
-            continue;
-        };
-        entry.unpack(full_install_path).unwrap();
+    let source_dir = get_cur_dir().join("libsodium/");
+    if !source_dir.join(".git").exists() {
```

Comment:
I believe we shouldn't call out to git here. If someone builds our crate from crates.io the source should be bundled (maybe there's something we need to add to the manifest to make it so) and if someone clones the repo the readme should tell them to clone with --recursive.